booster/library/Booster/SMT/Interface.hs

```diff
@@ -263,9 +263,9 @@ checkPredicates ::
     Set Predicate ->
     Map Variable Term ->
     Set Predicate ->
-    io (Either SMTError Bool)
+    io (Either SMTError (Maybe Bool))
```

Comment:
Since the `SMTSolverUnknown` is not a "real error", we could also use a 4-fold dedicated result type here
```
data CheckPredicatesResult
    = PredicatesTrue
    | PredicatesFalse
    | PredicatesIndeterminate
    | UnknownResult
```
That means we can throw exceptions directly from here as we did before. 
The current version puts the burden on the caller but leaves more flexibility.